docs/content/doc/usage/command-line.en-us.md

```diff
@@ -161,6 +161,38 @@ Admin operations:
         - `--restricted-group`: Group Claim value for restricted users. (Optional)
       - Examples:
         - `gitea admin auth update-oauth --id 1 --name external-github-updated`
+    - `add-smtp`:
+      - Options:
+        - `--name`: Application Name. Required.
+        - `--auth-type`: SMTP Authentication Type (PLAIN/LOGIN/CRAM-MD5). Default to PLAIN.
+        - `--host`: SMTP host. Required.
+        - `--port`: SMTP port. Required.
+        - `--force-smtps`: SMTPS is always used on port 465. Set this to force SMTPS on other ports.
+        - `--skip-verify`: Skip TLS verify.
+        - `--helo-hostname`: Hostname sent with HELO. Leave blank to send current hostname.
+        - `--disable-helo`: Disable SMTP helo.
+        - `--allowed-domains`: Leave empty to allow all domains. Separate multiple domains with a comma (',').
+        - `--skip-local-2fa`: Skip 2FA to log on.
+        - `--active`: This Authentication Source is Activated.
+      - Examples:
+        - `gitea admin auth add-smtp --name ldap --host smtp.mydomain.org --port 587 --skip-verify --active`
```

Comment:
The document differs (`--skip-verify` vs `--skip-verify=false`):

```
`gitea admin auth add-smtp --name ldap --host smtp.mydomain.org --port 587 --skip-verify --active`
```

```
`gitea admin auth update-smtp -id 1 --host smtp.mydomain.org --port 587 --skip-verify=false`
`gitea admin auth update-smtp -id 1 --active=false`
```
ps: the document says `--id` instead of the `-id` in the example.

Since most options are shared/common, maybe we can also group these options:

```
- `add-smtp`:
   - add-smtp/update-smtp shared options: 
      - .....
- `update-smtp`:
    - Options:
        - `--id`: ID of source to be updated. Required.
        - other options are shared with `add-smtp`
```

Reply:
> The document differs (`--skip-verify` vs `--skip-verify=false`):
> 
> ```
> `gitea admin auth add-smtp --name ldap --host smtp.mydomain.org --port 587 --skip-verify --active`
> ```

All bool options uses BoolTFlag which allows using `--option=[true|false]` format and `--option` defaults to true. If option is not specified than those option is not changed. See [github.com/urfave/cli](https://github.com/urfave/cli/issues/804).

Many of BoolFlag options in cli should be migrated to BoolTFlag as there is no way to disable those options. See my remark in https://github.com/go-gitea/gitea/pull/18197#discussion_r780680571

Reply:
So usually just not setting the option is the best way of indicating that the option is meant to be false but if BoolTFlag allows both then that's fine.

Reply:
> So usually just not setting the option is the best way of indicating that the option is meant to be false but if BoolTFlag allows both then that's fine.

Omitting option for false is fine in 'add-smtp'. But in 'update-smtp' that behavior would require specifying all enabled options on every call. With my approach you could just do `gitea admin auth update-smtp --id 1 --active=false` to deactivate provider without changing or specifying other options.

At first I replicated `add-oauth` and `update-oauth` approach but in [runUpdateOauth](https://github.com/go-gitea/gitea/blob/af9247392000c559bf8638f70b025d1894e6fd97/cmd/admin.go#L730) they also ended with "change needed options" instead of "change all on every call".  Their approach is fine for string options as you could specify --flag="" but not for bool options. As a example to implement `skip-local-2fa` option handling in `runUpdateOauth` we could:

1. Continue using BoolFlag() and 
```
if c.IsSet("skip-local-2fa") {
  oAuth2Config.SkipLocalTwoFA = c.Bool("skip-local-2fa")
}
```
which would not allow us to disable option as omitting options would just leave last value.

2. Use BoolFlag() and 
```
oAuth2Config.SkipLocalTwoFA = c.Bool("skip-local-2fa")
```
which would always set option. But omitting option would change it to false which would be contrary to approach for other options.

3. Use BoolTFlag() and 
```
if c.IsSet("skip-local-2fa") {
  oAuth2Config.SkipLocalTwoFA = c.BoolT("skip-local-2fa")
}
```
which would set option to value specified (either true or false) or leave option unchanged if omitted.

Reply:
ps: I meant that the documents differ for `add` and `update`.

For `add`: `--skip-verify --active` (and, does `--skip-verify` work with BoolT correctly?)
For `update`: `--skip-verify=false`

It confuses readers.

Reply:
Hmm ... the cli library does that trick, so --option is the same as --option=true.

👍 No question from my side.

Reply:
[Urfave cli library](github.com/urfave/cli) is really poorly documented. It took a while till I stumbled on right option.